Xamarin/DrawXShared/RealmDraw.cs

```diff
@@ -0,0 +1,552 @@
+        private float _canvasWidth, _canvasHeight;
+        private IList<DrawPath> _pathsToDraw = null;  // set in notification callback
+        #endregion
+
+        #region CachedCanvas
+        private int _canvasSaveCount;  // from SaveLayer
+        private bool _hasSavedBitmap = false;  // separate flag so we don't rely on any given value in _canvasSaveCount
+        private bool _redrawPathsAtNextDraw = true;
+        #endregion
+
+        #region Touch Areas
+        private SKRect _loginIconRect;
+        private SKRect _loginIconTouchRect;
+        //// setup in DrawBackground
+        private float _pencilWidth;
+        private float _pencilsTop;
+        private int _numPencils;
+        private List<SKBitmap> _pencilBitmaps;
+        private SKBitmap _loginIconBitmap;
+        #endregion
+
+        #region LoginState
+        private bool _waitingForLogin = false;
+        #endregion
+
+        #region Settings
+        private DrawXSettings Settings => DrawXSettingsManager.Settings;
+
+        private int _currentColorIndex = -1;  //// for quick check if pencil we draw is current color
+
+        private SwatchColor _currentColor;
+
+        private SwatchColor CurrentColor
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_currentColor.Name))
+                {
+                    InitCurrentColor();
+                }
+
+                return _currentColor;
+            }
+
+            set
+            {
+                if (!_currentColor.Name.Equals(value.Name))
+                {
+                    _currentColor = value;
+                    DrawXSettingsManager.Write(() => Settings.LastColorUsed = _currentColor.Name);
+                    _currentColorIndex = SwatchColor.Colors.IndexOf(_currentColor);
+                }
+            }
+        }
+
+        private void InitCurrentColor()
+        {
+            _currentColor = SwatchColor.ColorsByName [Settings.LastColorUsed];
+            _currentColorIndex = SwatchColor.Colors.IndexOf(_currentColor);
+        }
+        #endregion Settings
+
+        public RealmDraw(float inWidth, float inHeight)
+        {
+            // TODO close the Realm
+            _canvasWidth = inWidth;
+            _canvasHeight = inHeight;
+
+            // simple local open            
+            // _realm = Realm.GetInstance("DrawX.realm");
+
+            _pencilBitmaps = new List<SKBitmap>(SwatchColor.Colors.Count);
+            foreach (var swatch in SwatchColor.Colors)
+            {
+                _pencilBitmaps.Add(EmbeddedMedia.BitmapNamed(swatch.Name + ".png"));
+            }
+
+            _loginIconBitmap = EmbeddedMedia.BitmapNamed("CloudIcon.png");
+        }
+
+        internal void InvalidateCachedPaths()
+        {
+            _redrawPathsAtNextDraw = true;
+            _hasSavedBitmap = false;
+            _currentlyDrawing = null;
+        }
+
+        internal async void LoginToServerAsync()
```

Comment:
Hm, this method is really complex. I don't have time to look into it properly right now but it seems that an example like this is going to look scary where we want it to look simple.. :-(